artifactory/commands/transferfiles/manager.go

```diff
@@ -269,28 +270,31 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 		}
 		time.Sleep(waitTimeBetweenChunkStatusSeconds * time.Second)
 
-		// Run once per 3 minutes
-		if i%60 == 0 {
+		// Run once per 5 minutes
+		if i%100 == 0 {
 			// 'Working threads' are determined by how many upload chunks are currently being processed by the source Artifactory instance.
 			if err := phaseBase.stateManager.SetWorkingThreads(curProcessedUploadChunks); err != nil {
 				log.Error("Couldn't set the current number of working threads:", err.Error())
 			}
+			log.Debug("There are", len(phaseBase.stateManager.StaleChunks), "chunks in transit for more than 30 minutes")
+			log.Debug(fmt.Sprintf("Chunks in transit: %v", chunksLifeCycleManager.GetNodeIdToChunkIdsMap()))
 		}
 
 		// Each uploading thread receives a token and a node id from the source via the uploadChunkChan, so this go routine can poll on its status.
-		fillChunkDataBatch(&chunksLifeCycleManager, uploadChunkChan)
+		activeChunks := fillChunkDataBatch(&chunksLifeCycleManager, uploadChunkChan)
 		if err := chunksLifeCycleManager.StoreStaleChunks(phaseBase.stateManager); err != nil {
 			log.Error("Couldn't store the stale chunks:", err.Error())
 		}
 		// When totalChunks size is zero, it means that all the tokens are uploaded,
 		// we received 'DONE' for all of them, and we notified the source that they can be deleted from the memory.
 		// If during the polling some chunks data were lost due to network issues, either on the client or on the source,
 		// it will be written to the error channel
-		if chunksLifeCycleManager.totalChunks == 0 {
+		if activeChunks == 0 {
 			if shouldStopPolling(doneChan) {
 				log.Debug("Stopping to poll on uploads...")
 				return
 			}
+			log.Debug("Active chunks counter is 0, but we haven't done yet")
```

Comment:
...but we haven't done yet --> ...but the 'done' signal hasn't been receieved yet.